src/containerapp/azext_containerapp/custom.py

```diff
@@ -2370,6 +2374,57 @@ def delete_environment_telemetry_data_dog(cmd,
     return r
 
 
+def show_environment_telemetry_data_dog(cmd,
+                                        name,
+                                        resource_group_name):
+    raw_parameters = locals()
+
+    containerapp_env_telemetry_data_dog_decorator = ContainerappEnvTelemetryDataDogPreviewSetDecorator(
+        cmd=cmd,
+        client=ManagedEnvironmentPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    
+    containerapp_env_def = None
+    try:
+        containerapp_env_def = containerapp_env_telemetry_data_dog_decorator.show()
+    except Exception as e:
+        handle_raw_exception(e)
+
+    if not containerapp_env_def:
+        raise ResourceNotFoundError("The containerapp environment '{}' does not exist".format(name))
+
+    try:
+        r = containerapp_env_def["properties"]["openTelemetryConfiguration"]["destinationsConfiguration"]["dataDogConfiguration"]
+        if r is None:
+            raise ValidationError("The containerapp environment '{}' does not have data dog enabled.".format(name)) from e
+        r = {}
+
+        if "key" in containerapp_env_def["properties"]["openTelemetryConfiguration"]["destinationsConfiguration"]["dataDogConfiguration"]:
+            safe_set(r, "dataDogConfiguration", "key", value=DEFAULT_CONFIGURED_STR)
+
+        safe_set(r, "dataDogConfiguration", "site", value=safe_get(containerapp_env_def, "properties", "openTelemetryConfiguration", "destinationsConfiguration", "dataDogConfiguration", "site"))
+
+        enable_open_telemetry_traces = False
+        existing_traces = safe_get(containerapp_env_def, "properties", "openTelemetryConfiguration", "tracesConfiguration", "destinations")
+        if existing_traces and DATA_DOG_DEST in existing_traces:
+            enable_open_telemetry_traces = True
+        
+        safe_set(r, "dataDogConfiguration", "enable-open-telemetry-traces", value=enable_open_telemetry_traces)
```

Comment:
>enable-open-telemetry-traces [](http://example.com/codeflow?start=45&length=28)

??  with - in property?

Reply:
yes
the response will look like this
{
  "dataDogConfiguration": {
    "enable-open-telemetry-metrics": true,
    "enable-open-telemetry-traces": true,
    "key": "Configured",
    "site": "datadoghq.com"
  }
}